lua/lspconfig/server_configurations/zk.lua

```diff
@@ -7,17 +7,21 @@ return {
     root_dir = util.root_pattern '.zk',
   },
   commands = {
```

Comment:
Feels like such a breaking removal would belong in a separate PR?

Reply:
Is the deprecation still being discussed or has it been already decided?

I am neither a maintainer nor a significant contributor  so I dont know if my opinion counts for anything, but for what its wroth:
I guess for certain servers features (like `clangd/switchSourceHeader`) these are considered as "extended" features, and certainly off 3.16 spec. 
But for servers like `texlab`, where the buffer holds more than 80% plain text and very few lines of latex commands, `texlab/build` is basically a primary feature. 

Without that command internally defined, users will have to resort to defining `params`, making the lsp call, extracting the results contextually and displaying them,  and then writing a `user_command` to call it. Which most users may not be able to do. Kind of removes the quick out of quickstart feature of the plugin. 

Unless someone writes a plugin for just `texlab/build` and `texlab/forward` commands. And such plugins for each server that has those command features.

Of course, if that is the consensus I will remove/or deprecate it!

Reply:
> Is the deprecation still being discussed or has it been already decided?
>
> I am neither a maintainer nor a significant contributor so I dont know if my opinion counts for anything, but for what its wroth:

Likewise. If by deprecation you mean these commands for `zk` then no, I don't believe it has been discussed before (quick GitHub search).

My take on this is: I really don't see the point of removing commands just for the sake of nudging users to install a dedicated plugin (https://github.com/mickael-menu/zk-nvim) for it (even though that plugin looks very thorough and well-made). It just adds unnecessary friction. If there are people that rely on these commands it's obviously providing value to them, and I think that weighs heavier than how to technically slice and dice things. If the commands ever stop working, or if they no longer suffice, these users have a very strong personal incentive to switch to https://github.com/mickael-menu/zk-nvim themselves.

Reply:
My suggestion would be to keep the internals, which are defined in `server_configurations/[lsp].lua` but deprecate the interface for defining them in setup as @justinmk  recommended. 

It is very trivial to just define custom commands in `nvim_create_user_command` instead of having`lspconfig` scan the the configs and define them using the same interface.

Reply:
Sounds extremely reasonable to me!